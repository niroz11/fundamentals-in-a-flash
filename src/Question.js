import React, { Component } from 'react';

class Question extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  saveCorrectlyAnsweredIdToStorage = (questionId, questionCategory) => {
    let correctIds = [];
    if (localStorage.hasOwnProperty(questionCategory)) {
      correctIds = JSON.parse(localStorage.getItem(questionCategory));
    }
    correctIds.push(questionId);
    correctIds = JSON.stringify(correctIds);
    localStorage.setItem(questionCategory, correctIds);
  }

  validateIfCorrectAnswer = (event) => {
    event.preventDefault();
    if (event.target.innerHTML === this.props.currentQuestion.correct_answer) {
      this.saveCorrectlyAnsweredIdToStorage(this.props.currentQuestion.id, this.props.currentQuestion.category)
      this.props.updateCorrectCounter()
      this.props.updateUserStatus('correct')
    } else {
      this.props.updateUserStatus('wrong')
    }
  }

  render() {
    let { id, category, question, answers, correct_answer, resources } = this.props.currentQuestion;
    let style = category.toLowerCase().split(' ').join('-')
    return (
      <div className={"question-contain " + style}>
        <p className="question">{question}</p>
        {
          answers.map((answer, index) => {
            return <button onClick={this.validateIfCorrectAnswer} key={'answer' + index}>{answer}</button>
          })
        }
      </div>
    )
  }
}

export default Question;