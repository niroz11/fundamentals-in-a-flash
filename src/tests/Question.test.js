import React from 'react';
import Question from '../Question';
import { shallow } from 'enzyme';

const updateUserStatusMock = jest.fn();
const questionMock = {
  id: 0,
  category: "Prototype methods",
  question: "What does the forEach() method return?",
  answers: ["undefined", "an array of the same length with all elements modified", "the original array with its elements modified"],
  correct_answer: "undefined",
  resources: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach"
}

describe('Question', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Question
        currentQuestion={questionMock}
        updateUserStatus={updateUserStatusMock}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});