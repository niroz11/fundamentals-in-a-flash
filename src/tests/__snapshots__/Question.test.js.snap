// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match the snapshot with all data passed in 1`] = `
<div
  className="question-contain prototype-methods"
>
  <p>
    Question 
    2
     of 
    8
  </p>
  <p
    className="question"
  >
    What does the forEach() method return?
  </p>
  <button
    className="answer0"
    key="answer0"
    onClick={[Function]}
  >
    undefined
  </button>
  <button
    className="answer1"
    key="answer1"
    onClick={[Function]}
  >
    an array of the same length with all elements modified
  </button>
  <button
    className="answer2"
    key="answer2"
    onClick={[Function]}
  >
    the original array with its elements modified
  </button>
  <button
    className="skip-btn"
    onClick={[MockFunction]}
  >
    Skip
  </button>
</div>
`;
